contracts/README.md

```diff
@@ -52,3 +52,7 @@ The `totalNativeAssetSupply` implementation of `NativeTokenDestination` takes in
 - the number of native tokens that it has minted
 - the number of native tokens that have been burned to pay for transaction fees
 - the number of native tokens "burned" to be bridged back to the home chain, which are sent to a pre-defined `BURNED_FOR_BRIDGE_ADDRESS`.
+
+# Teleporter Message Fees
+
+Fees can be optionally added to Teleporter messages in order to incentive relayers to deliver them, as documented [here](https://github.com/ava-labs/teleporter/tree/main/contracts/src/Teleporter#fees). The bridge contracts in this repository allow for specifying any ERC20 token and amount to be used as the Teleporter message fee for single-hop transfers between `TeleporterTokenSource` and `TeleporterTokenDestination` instances. Multi-hop transfers between two `TeleporterTokenDestination` instances involve two Teleporter messages: the first from the initiating chain to source chain, and the second from the source chain on to the final destination. In the multi-hop case, the first message fee can be paid in any ERC20 token and amount (similar to the single-hop case), but the second message fee must be paid in-kind of the asset being transferred and is deducted from the amount being bridged. This restriction on the secondary message fee is necessary because the transaction on the source chain routing the funds to the destination chain is not sent by the wallet performing the transfer. Because of this, it can not directly spend an arbitrary ERC20 token from that wallet. Using the asset being transferred for the optional secondary fee allows user's to perform an incentivized multi-hop transfer without needing to make any interaction with the source chain themselves. If there is a need for the second message from the source chain to the final destination chain to pay a fee in another asset, it is recommended to perform two single-hop transfers, which allows for specifying an arbitrary ERC20 token to be used for the fee of each.
```

Comment:
```suggestion
Fees can be optionally added to Teleporter messages in order to incentivize relayers to deliver them, as documented [here](https://github.com/ava-labs/teleporter/tree/main/contracts/src/Teleporter#fees). The bridge contracts in this repository allow for specifying any ERC20 token and amount to be used as the Teleporter message fee for single-hop transfers between `TeleporterTokenSource` and `TeleporterTokenDestination` instances. Multi-hop transfers between two `TeleporterTokenDestination` instances involve two Teleporter messages: the first from the initiating chain to source chain, and the second from the source chain on to the final destination. In the multi-hop case, the first message fee can be paid in any ERC20 token and amount (similar to the single-hop case), but the second message fee must be paid in-kind of the asset being transferred and is deducted from the amount being bridged. This restriction on the secondary message fee is necessary because the transaction on the source chain routing the funds to the destination chain is not sent by the wallet performing the transfer. Because of this, it can not directly spend an arbitrary ERC20 token from that wallet. Using the asset being transferred for the optional secondary fee allows users to perform an incentivized multi-hop transfer without needing to make any interaction with the source chain themselves. If there is a need for the second message from the source chain to the final destination chain to pay a fee in another asset, it is recommended to perform two single-hop transfers, which allows for specifying an arbitrary ERC20 token to be used for the fee of each.
```